File: frontend/types/community.ts
// frontend/types/community.ts
/**
 * 互助社区类型定义
 */

export type ChannelType = 'professional' | 'experience';

export type UserRole =
  | 'mom'
  | 'dad'
  | 'family'
  | 'certified_doctor'
  | 'certified_therapist'
  | 'certified_nurse';

export type ContentStatus =
  | 'draft'
  | 'pending_review'
  | 'published'
  | 'hidden';

export interface Author {
  id: string;
  nickname: string;
  avatar_url: string | null;
  role: UserRole;
  is_certified: boolean;
  certification_title?: string; // e.g., "北京协和医院 妇产科 主任医师"
}

export interface Tag {
  id: string;
  name: string;
  slug: string;
}

export interface Question {
  id: string;
  title: string;
  content: string;
  content_preview: string;
  channel: ChannelType;
  status: ContentStatus;
  author: Author;
  tags: Tag[];
  image_urls: string[];
  view_count: number;
  answer_count: number;
  like_count: number;
  collection_count: number;
  is_liked: boolean;
  is_collected: boolean;
  professional_answer_count: number;
  experience_answer_count: number;
  created_at: string;
  has_accepted_answer: boolean;
}

export interface Answer {
  id: string;
  question_id: string;
  author: Author;
  content: string;
  content_preview: string;
  image_urls: string[];
  is_professional: boolean;
  is_accepted: boolean;
  like_count: number;
  comment_count: number;
  is_liked: boolean;
  created_at: string;
}

export interface Comment {
  id: string;
  answer_id: string;
  author: Author;
  content: string;
  parent_id: string | null;
  reply_to_user: Author | null;
  like_count: number;
  is_liked: boolean;
  created_at: string;
  replies?: Comment[];
}

export interface HotTopic {
  id: string;
  name: string;
  question_count: number;
  trend: 'up' | 'down' | 'stable';
}

// 角色配置
export const ROLE_CONFIG: Record<UserRole, { label: string; badgeColor: string; icon?: string }> = {
  mom: { label: '妈妈', badgeColor: 'bg-pink-100 text-pink-700' },
  dad: { label: '爸爸', badgeColor: 'bg-blue-100 text-blue-700' },
  family: { label: '家属', badgeColor: 'bg-stone-100 text-stone-600' },
  certified_doctor: {
    label: '认证医生',
    badgeColor: 'bg-emerald-100 text-emerald-700',
    icon: '✓'
  },
  certified_therapist: {
    label: '认证康复师',
    badgeColor: 'bg-teal-100 text-teal-700',
    icon: '✓'
  },
  certified_nurse: {
    label: '认证护士',
    badgeColor: 'bg-cyan-100 text-cyan-700',
    icon: '✓'
  },
};

// 频道配置
export const CHANNEL_CONFIG: Record<ChannelType, {
  label: string;
  subtitle: string;
  color: { bg: string; text: string; accent: string };
}> = {
  professional: {
    label: '专业频道',
    subtitle: '听听医生怎么说',
    color: {
      bg: 'bg-sky-50',
      text: 'text-sky-700',
      accent: '#0ea5e9', // sky-500
    },
  },
  experience: {
    label: '经验频道',
    subtitle: '看看妈妈们的经验',
    color: {
      bg: 'bg-amber-50',
      text: 'text-amber-700',
      accent: '#f59e0b', // amber-500
    },
  },
};

File: backend/internal/model/community.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

// ChannelType enum
type ChannelType string

const (
	ChannelProfessional ChannelType = "professional"
	ChannelExperience   ChannelType = "experience"
)

// ContentStatus enum
type ContentStatus string

const (
	StatusDraft         ContentStatus = "draft"
	StatusPendingReview ContentStatus = "pending_review"
	StatusPublished     ContentStatus = "published"
	StatusHidden        ContentStatus = "hidden"
	StatusDeleted       ContentStatus = "deleted"
)

type Question struct {
	ID       string `gorm:"type:varchar(36);primaryKey" json:"id"`
	AuthorID string `gorm:"type:varchar(36);index;not null" json:"author_id"`

	// Content
	Title     string  `gorm:"type:varchar(200);index;not null" json:"title"`
	Content   string  `gorm:"type:text;not null" json:"content"`
	ImageURLs *string `gorm:"type:text" json:"image_urls"` // JSON array

	// Channel
	Channel ChannelType `gorm:"type:varchar(20);index;default:'experience'" json:"channel"`

	// Status
	Status ContentStatus `gorm:"type:varchar(20);index;default:'pending_review'" json:"status"`

	// Statistics
	ViewCount       int `gorm:"default:0" json:"view_count"`
	AnswerCount     int `gorm:"default:0" json:"answer_count"`
	LikeCount       int `gorm:"default:0" json:"like_count"`
	CollectionCount int `gorm:"default:0" json:"collection_count"`

	// Flags
	IsPinned   bool `gorm:"default:false" json:"is_pinned"`
	IsFeatured bool `gorm:"default:false" json:"is_featured"`

	// Accepted answer
	AcceptedAnswerID *string `gorm:"type:varchar(36)" json:"accepted_answer_id"`

	// Timestamps
	CreatedAt   time.Time  `gorm:"index" json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
	PublishedAt *time.Time `json:"published_at"`

	// Relationships
	Author  User     `gorm:"foreignKey:AuthorID" json:"author,omitempty"`
	Answers []Answer `gorm:"foreignKey:QuestionID" json:"answers,omitempty"`
	Tags    []Tag    `gorm:"many2many:question_tags" json:"tags,omitempty"`
}

func (q *Question) BeforeCreate(tx *gorm.DB) error {
	if q.ID == "" {
		q.ID = uuid.New().String()
	}
	return nil
}

type Answer struct {
	ID         string `gorm:"type:varchar(36);primaryKey" json:"id"`
	QuestionID string `gorm:"type:varchar(36);index;not null" json:"question_id"`
	AuthorID   string `gorm:"type:varchar(36);index;not null" json:"author_id"`

	// Content
	Content   string  `gorm:"type:text;not null" json:"content"`
	ImageURLs *string `gorm:"type:text" json:"image_urls"`

	// Author role info
	AuthorRole     UserRole `gorm:"type:varchar(30);index" json:"author_role"`
	IsProfessional bool     `gorm:"index;default:false" json:"is_professional"`

	// Expert post: only certified professionals can publish
	IsExpertPost bool    `gorm:"index;default:false" json:"is_expert_post"`
	Sources      *string `gorm:"type:text" json:"sources"` // Required for expert posts

	// Status
	Status ContentStatus `gorm:"type:varchar(20);index;default:'pending_review'" json:"status"`

	// Statistics
	LikeCount    int `gorm:"default:0" json:"like_count"`
	CommentCount int `gorm:"default:0" json:"comment_count"`

	// Accepted
	IsAccepted bool `gorm:"default:false" json:"is_accepted"`

	// Timestamps
	CreatedAt time.Time `gorm:"index" json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`

	// Relationships
	Author   User      `gorm:"foreignKey:AuthorID" json:"author,omitempty"`
	Question Question  `gorm:"foreignKey:QuestionID" json:"-"`
	Comments []Comment `gorm:"foreignKey:AnswerID" json:"comments,omitempty"`
}

func (a *Answer) BeforeCreate(tx *gorm.DB) error {
	if a.ID == "" {
		a.ID = uuid.New().String()
	}
	return nil
}

type Comment struct {
	ID       string `gorm:"type:varchar(36);primaryKey" json:"id"`
	AnswerID string `gorm:"type:varchar(36);index;not null" json:"answer_id"`
	AuthorID string `gorm:"type:varchar(36);index;not null" json:"author_id"`

	// Nested reply
	ParentID      *string `gorm:"type:varchar(36)" json:"parent_id"`
	ReplyToUserID *string `gorm:"type:varchar(36)" json:"reply_to_user_id"`

	// Content
	Content string `gorm:"type:text;not null" json:"content"`

	// Status
	Status ContentStatus `gorm:"type:varchar(20);default:'pending_review'" json:"status"`

	// Statistics
	LikeCount int `gorm:"default:0" json:"like_count"`

	// Timestamps
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`

	// Relationships
	Author      User   `gorm:"foreignKey:AuthorID" json:"author,omitempty"`
	Answer      Answer `gorm:"foreignKey:AnswerID" json:"-"`
	ReplyToUser *User  `gorm:"foreignKey:ReplyToUserID" json:"reply_to_user,omitempty"`
}

func (c *Comment) BeforeCreate(tx *gorm.DB) error {
	if c.ID == "" {
		c.ID = uuid.New().String()
	}
	return nil
}
